web/src/routes/execution.tsx

```diff
@@ -0,0 +1,134 @@
+import { Button, Card, Container } from "react-bootstrap"
+import QRCode from "react-qr-code"
+import { ActionFunctionArgs, LoaderFunctionArgs, useActionData, useLoaderData, useParams } from "react-router"
+import { isType } from "../utils"
+import { useEffect, useState } from "react"
+import { getExecutionStatus, startExecution, stopExecution } from "../api"
+import _ from "lodash"
+import { config } from "../config"
+import { Form } from "react-router-dom"
+import { CsrfInput } from "../contexts/csrf"
+
+function TanCard({ tan }: { tan: string }) {
+  return (
+    <Card className="d-flex m-1">
+      <QRCode value={tan} className="align-self-center p-3" />
+      <Card.Body>
+        <Card.Title className="text-center">{tan}</Card.Title>
+      </Card.Body>
+    </Card >
+  )
+}
+
+interface Player {
+  tan: string,
+  name: string,
+  status: string,
+  action: string
+}
+
+interface ExecutionData {
```

Comment:
I would recommend to relocate the interface in the file. In here between the functions it seems a little lost.